aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -11,6 +11,42 @@
 namespace at {
 namespace native {
 
+Tensor quantize_per_tensor_dynamic(
+    const Tensor& self,
+    ScalarType dtype,
+    bool reduce_range) {
+  auto input_contig = self.contiguous();
+  float x_min = input_contig.min().item<float>();
+  float x_max = input_contig.max().item<float>();
+
+  if (reduce_range && at::globalContext().qEngine() == at::QEngine::QNNPACK) {
+    reduce_range = false;
+  }
+
+  int qmin = 0;
+  int qmax = 255;
+
+  // should we check additional QInt types beyond 8 bits?
+  if (dtype == ScalarType::QInt8) {
+    qmin = -128;
+    qmax = 127;
+  } else if (dtype == ScalarType::QUInt8) {
+  } else {
+    // dtype not supported
+  }
```

Comment:
I was considering doing the same as https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_tensor.py#L31, but in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/quant_utils.h#L70, the `reduce_flag` is processed, so I think since I'm calling quant_utils::ChooseQuantizationParams, then I don't need to manually process the reduce_flag?

Reply:
oh I see, yeah that sounds good, please put qmin/qmax assignment for QUInt8 to the else branch though for better readability.

Also we can add a `TORCH_ASSERT` for the dtype: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/conv_serialization.h#L284